### PR TITLE
docs: ADR for alembic migrations

### PR DESCRIPTION
**Pull request by Ian2012**
## Description

This PR adds documentation about the alembic migrations tool

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: docs/decisions/0007_clickhouse_migrations.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Accepted
+
+Context
+*******
+
+Alembic is a migration tool for SQLAlchemy. It is used to manage the clickhouse database
+schema for the Aspects project.
+
+Alembic has support for branching migrations and multiple databases, it creates a migrations
```
Comment (bmtcril) [inline docs/decisions/0007_clickhouse_migrations.rst:15]: Can we add that it supports migration rollbacks too?

**Comment by Ian2012** [comment]
@bmtcril ready

File: docs/decisions/0007_clickhouse_migrations.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   create the databases, users and permissions, and any other necessary setup before
+   running migrations.
+#. Migrations are executed with RAW SQL using the ``op.execute`` method. This allows
+   us to use ClickHouse-specific SQL syntax.
+#. Migrations should always define a downgrade step, even if it is a no-op.
+#. Migrations are generated using a Sequential ID. This allows us to easily identify
+   the order in which migrations were created and executed.
+#. Migrations cannot be modified once they are released to production. If a migration
+   needs to be modified, a new migration should be created with a new sequential ID.
+#. The alembic migrations table is highly couple with the xAPI database, for this the
```
[the remark this answers is not in the document]
Comment (Ian2012) [reply docs/decisions/0007_clickhouse_migrations.rst]: I like it, it's better than my approach and would reflect the project's progress better

**Review by bmtcril — APPROVED** [review]
Thanks!

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-aspects&entry.867055334=https://github.com/openedx/openedx-aspects/pull/56&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-07-05+12:36&entry.1917517419=2023-07-05+14:50&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**